Software/Development/SemesterProject/App/openLoopControlManager.c

```diff
@@ -36,44 +41,106 @@ int16 mapInt(int16 x, int16 xMax, int16 yMax)
 }
 
 
-
-void setConstantDutyFromPot(void)
+/*
+ * This function sets all the PWM signals to a constant value
+ * obtained from a reference.
+ */
+void setConstantDutyFromReference(void)
 {
-    setAllDuties(mapFloat(1.2, 3.3, readMaxDuty())); //mapFloat(readADC(), readMaxADC(), readMaxDuty()));
+    float duty;
+    float torqueRef, maxRef;
+    /*  INCLUDE THIS WHEN EVERYTHING IS ADDED
+    switch (getReferenceInput())
+    {
+        case POT:
+        {
+            torqueRef = getTorqueReferenceSliderMeasurement();
+            maxRef = getMaximumReferenceValue();
+        }break;
+        case GUI:
+        {
+            torqueRef = getTorqueReferenceFromGUI();
+            maxRef = getMaximumReferenceGUI();
+        }
+        case PEDAL:
+        {
+            torqueRef = getTorqueReferenceFromPedal();
+            maxRef = getMaximumReferencePedal();
+        }
+
+    }*/
+
+    torqueRef = getTorqueReferenceSliderMeasurement();
+    maxRef = 3.3;//getMaximumReferenceADC();
+
+    duty = mapFloat(torqueRef, maxRef, readMaxDuty());
+
+    setAllDuties(duty);
 }
 
 
 //
-//This function is called at switching frequency so the duty needs to get updated in every iteration.
+//Function called at switching frequency.
+//It sets VF control on the motor according to the torque reference potentiometer.
 //
-void setFrequencyFromPot(void)
+void openLoopVFControl(void)
```

Comment:
As a general statement, functions should be short. In clean code they say:

> How short should a function be? In 1999 I went to visit Kent Beck at his home in Oregon. We sat down and did some programming together. At one point he showed me a cute little Java/Swing program that he called Sparkle. It produced a visual effect on the screen very similar to the magic wand of the fairy godmother in the movie Cinderella. As you moved the mouse, the sparkles would drip from the cursor with a satisfying scintillation, falling to the bottom of the window through a simulated gravitational field. When Kent showed me the code, I was struck by how small all the functions were. I was used to functions in Swing programs that took up miles of vertical space. Every function in this program was just two, or three, or four lines long. Each was transparently obvious. Each told a story. And each led you to the next in a compelling order. That’s how short your functions should be!

In this kind of function, which is doing so many things, you might reach let's say 10 or 12 and all the tasks should be in the same level of abstraction (as mentioned in previous review). The function is quite massive at the moment. Nice name though.

Reply:
Take a look now.